clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -749,6 +749,7 @@ public boolean exists(Path path) throws IOException {
         }
         // use listing to check if directory exists
         ListingIterator iterator = new ListingIterator(path, true, 1);
+        iterator.setRemoveDirectory(false);
```

Comment:
Good catch!

This is *obviously* a bad move for a readonly operation such as `exists`.  Do you know why we had this in the first place?